src/mpid/ch4/shm/xpmem/xpmem_init.c

```diff
@@ -26,8 +26,26 @@ int MPIDI_XPMEM_mpi_init_hook(int rank, int size, int *n_vcis_provided, int *tag
     /* Try to share entire address space */
     MPIDI_XPMEM_global.segid = xpmem_make(0, XPMEM_MAXADDR_SIZE, XPMEM_PERMIT_MODE,
                                           MPIDI_XPMEM_PERMIT_VALUE);
-    /* 64-bit segment ID or failure(-1) */
-    MPIR_ERR_CHKANDJUMP(MPIDI_XPMEM_global.segid == -1, mpi_errno, MPI_ERR_OTHER, "**xpmem_make");
+
+    /* Check to see if all processes were able to start up XPMEM. */
+    mpi_errno = MPIDU_Init_shm_allsum(MPIDI_XPMEM_global.segid == -1, &ret);
+    MPIR_ERR_CHECK(mpi_errno);
```

Comment:
It might make more sense if we use `logical and` here.

Reply:
I'm not sure what you mean. This is just a regular `mpi_errno` value. The "return value" of the previous function is the `ret` variable checked on the next line.

Reply:
I meant it is not straightforward what the sum of `MPIDI_XPMEM_global.segid == -1` means. Thus I suggested `logical add` as the reduce operation.

Reply:
Ah. I see. That makes more sense. I guess it comes down to the decision from the other discussion.